src/main/java/org/spongepowered/api/command/conversation/ConversationArchetype.java

```diff
@@ -0,0 +1,351 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command.conversation;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+/**
+ * A reusable collection of information which is used to
+ * create a {@link Conversation}.
+ */
+public interface ConversationArchetype {
+
+    /**
+     * Creates a new {@link ConversationArchetype.Builder} to build a {@link ConversationArchetype}.
+     *
+     * @return The new builder
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Gets the id for the archetype.
+     *
+     * @return The id
+     */
+    String getId();
+
+    /**
+     * Gets the message which will be sent to the conversants when the
+     * conversation starts, if it is set.
+     *
+     * @return The starting message, if available
+     */
+    Optional<Text> getStartingMessage();
+
+    /**
+     * Gets the question the conversation will start with.
+     *
+     * @return The first question
+     */
+    Question getFirstQuestion();
+
+    /**
+     * Gets if the conversation will catch the {@link Conversant}'s output.
+     *
+     * @return Whether or not the conversation will catch the {@link Conversant}'s output
+     */
+    boolean catchesOutput();
+
+    /**
+     * Gets if {@link Conversant}s can use commands.
+     *
+     * @return Whether or not {@link Conversant}s can use commands
+     */
+    boolean allowsCommands();
+
+    /**
+     * Gets the default external chat handler type for all {@link Conversant}s
+     * within the conversation.
+     *
+     * @return The default external chat handler type for all {@link Conversant}s
+     *     within the conversation
+     */
+    ExternalChatHandlerType getDefaultChatHandlerType();
+
+    /**
+     * Gets the handlers that handles the end of the {@link Conversation}.
+     *
+     * <p>The collection is implementation dependent, but is
+     * generally immutable or a copy.</p>
+     *
+     * @return All ending handlers
+     */
+    Collection<EndingHandler> getEndingHandlers();
```

Comment:
Given `Set` implements `Collection`, this is a moot point. Why would a plugin care? They would just want all the ending handlers. The _implementation_ should be what decides this, not the API.